### PR TITLE
Add validity checks for url, cfpUrl, twitter and location online check

### PR DESCRIPTION
**Pull request by cgrail**

### CODE REVIEW

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 1]
```diff
@@ -402,7 +418,7 @@ const ConferenceNewPage: React.FC = () => {
                   value={twitter}
                   onChange={handleFieldChange}
                 />
-                {errorFor('twitter', 'Twitter handle is required.')}
+                {errorFor('twitter', 'Twitter handle should be valid and formatted like @twitter')}
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:421]:
```suggestion
                {errorFor('twitter', 'Should be formatted like @twitter')}
```

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 2]
```diff
@@ -379,7 +395,7 @@ const ConferenceNewPage: React.FC = () => {
                     value={cfpUrl}
                     onChange={handleFieldChange}
                   />
-                  {errorFor('cfpUrl', 'CFP URL is required.')}
+                  {errorFor('cfpUrl', 'CFP URL should be valid. No URL query parameters or URL shorteners are allowed.')}
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:398]:
```suggestion
                  {errorFor('cfpUrl', 'No URL query parameters or URL shorteners are allowed.')}
```

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 3]
```diff
@@ -293,7 +307,7 @@ const ConferenceNewPage: React.FC = () => {
                     id='url'
                     onChange={handleFieldChange}
                   />
-                  {errorFor('url', 'Url is required.')}
+                  {errorFor('url', 'Url is required and should be valid. No URL query parameters or URL shorteners are allowed.')}
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:310]:
```suggestion
                  {errorFor('url', 'No URL query parameters or URL shorteners are allowed.')}
```

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 4]
```diff
@@ -85,18 +85,32 @@ const ConferenceNewPage: React.FC = () => {
...
+    const onlineRegex = /online|remote|everywhere|world|web|global|virtual|www|http/i
+    const isNotOnline = locationType !== 'online'
+    const httpRegex = /^http(s?):\/\//
+    const httpNoQuestionmarkRegex = /\?/
+    const urlShortener = /(\/bit\.ly)|(\/t\.co)/
+    const twitterRegex = /@(\w){1,15}$/
+
+    function isUrlValid(url:string) {
+      return httpRegex.test(url) && !httpNoQuestionmarkRegex.test(url) && !urlShortener.test(url)
+    }
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:98]: // prettier
```suggestion
    function isUrlValid(url: string) {
      return (
        httpRegex.test(url) &&
        !httpNoQuestionmarkRegex.test(url) &&
        !urlShortener.test(url)
      )
    }
```

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx (hunk 1 above)
Comment (nimzco) [reply src/components/ConferenceNewPage/ConferenceNewPage.tsx:421]: We should remove the initial state on line 51 or allow sending `"@"` as I strip it on the backend.

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 5]
```diff
@@ -85,18 +85,32 @@ const ConferenceNewPage: React.FC = () => {
...
-    const { topic, startDate, endDate, city, country, name, url } = conference
+    const { topic, startDate, endDate, city, country, name, url, cfpUrl, twitter } = conference
+    const onlineRegex = /online|remote|everywhere|world|web|global|virtual|www|http/i
+    const isNotOnline = locationType !== 'online'
+    const httpRegex = /^http(s?):\/\//
+    const httpNoQuestionmarkRegex = /\?/
+    const urlShortener = /(\/bit\.ly)|(\/t\.co)/
+    const twitterRegex = /@(\w){1,15}$/
+
+    function isUrlValid(url:string) {
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:96]: You can move all these declaration (except `isNotOnline`) on line 68, before the component declaration. You can also make them `CONSTANTS`.

**Review by nimzco — COMMENTED** [review]
Ha, I actually thought about validating these fields yesterday and I had added a reminder for March! Thanks for doing this, it's a great idea. I do some validation on the backend, but doing it here is better indeed.

Anyway, left a few comments to shorten sentences and a few nitpicks, otherwise 👌🏼

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx (hunk 5 above)
Comment (cgrail) [reply src/components/ConferenceNewPage/ConferenceNewPage.tsx]: done

**Comment by cgrail** [comment]
@nimzco thanks a lot. I've incorporated them.

**Comment by nimzco** [comment]
🙌🏼

**Pull request merged**